extensions/iceberg/src/main/java/io/deephaven/iceberg/location/IcebergTableLocationKey.java

```diff
@@ -0,0 +1,70 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.iceberg.location;
+
+import io.deephaven.engine.table.impl.locations.TableLocationKey;
+import io.deephaven.engine.table.impl.locations.local.URITableLocationKey;
+import io.deephaven.parquet.table.ParquetInstructions;
+import io.deephaven.parquet.table.location.ParquetTableLocationKey;
+import org.apache.iceberg.FileFormat;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.net.URI;
+import java.util.Map;
+
+/**
+ * {@link TableLocationKey} implementation for use with data stored in the parquet format.
+ */
+public class IcebergTableLocationKey extends URITableLocationKey {
+
+    private static final String IMPLEMENTATION_NAME = IcebergTableLocationKey.class.getSimpleName();
+
+    final FileFormat format;
+    final URITableLocationKey internalTableLocationKey;
+
+    /**
+     * Construct a new IcebergTableLocationKey for the supplied {@code fileUri} and {@code partitions}.
+     *
+     * @param fileUri The file that backs the keyed location.
```

Comment:
Please be consistent with formatting. (Sorry, I see that you inherited this through copy-paste. You still get to own it.)